squidDrone/src/com/com_message_buffer.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef SRC_COM_MESSAGE_BUFFER_HPP_
+#define SRC_COM_MESSAGE_BUFFER_HPP_
+
+#include <array>
+#include <queue>
+#include "com_types.hpp"
+
+namespace com {
+  /** 
+    * @brief This class represents a queue style buffer for communication
+    * frames exchanged between drones.
+    * 
+    */
+  class ComMessageBuffer {
+    public:
+      /** 
+       * @brief Construct a new Com Message Buffer object
+       * 
+       */
+      ComMessageBuffer(){};
+
+      /** 
+       * @brief Transfer data to queue buffer. 
+       * 
+       * @param data Reference to data frame. Must conform to data frame specifications.
+       * @return types::ComError COM_OK if ok, COM_BUFFER_IO_ERROR if not ok.
+       * 
+       */
+      auto PutData(std::array<std::uint8_t, 32> &data) noexcept -> types::ComError;
+
+      /** 
+       * @brief Retrieve data frame from queue buffer.
+       * 
+       * @return std::array<std::uint8_t, 32> 
+       * 
+       */
+      auto GetData() noexcept -> std::array<std::uint8_t, 32>;
+
+      /**
+       * @brief Check if the buffer is empty.
+       * 
+       * @return true If buffer is empty returns true.
+       * @return false If buffer is not empty returns false.
+       */
+      auto BufferIsEmpty() const noexcept -> bool;
+    
+    protected:
+      /** 
+       * @brief Queue to hold the 32 byte long data frames. Queue shall be emptied on
+       * each execution slice and the data frames shall be processed.
+       * 
+       */
+      std::queue<std::array<std::uint8_t,32>> data_;
```

Comment:
why is this protected and not private? You don't expect to inherit from this class in production code.

Reply:
Considered this a valid method to create a defined state inside the queue for testing purposes.

Reply:
Testing purposes should not change the memory layout of the production code. If you choose to make it protected and not private, this changes the memory layout.

If you have to absolutely have to use private members for testing. You can either befriend the testing class or temporarely #define private public in your test. Of course, this destroys encapsulation and is evil. But for testing, it serves the purpose.
Every other option changes the production code in a major way and that is to be prevented at all cost

Reply:
Understood! Fixed with 7ad53cc798ba88f5254fe13be7e3c41cf32165f6